src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -1428,7 +1474,29 @@ class ArchiveBuilder::CDSMapLogger : AllStatic {
         int array_len = arrayOop(source_oop)->length();
         st->print_cr("%s length: %d", source_oop->klass()->external_name(), array_len);
       } else {
-        st->print_cr("%s", source_oop->klass()->external_name());
+        st->print("%s", source_oop->klass()->external_name());
+
+        if (java_lang_String::is_instance(source_oop)) {
+          st->print(" ");
+          java_lang_String::print(source_oop, st);
+        } else if (java_lang_Class::is_instance(source_oop)) {
+          oop scratch_mirror = source_oop;
+
+          st->print(" ");
+          print_class_signature_for_mirror(st, scratch_mirror);
+
+          if (print_aot_init) {
+            Klass* src_klass = java_lang_Class::as_Klass(scratch_mirror);
+            if (src_klass != nullptr && src_klass->is_instance_klass()) {
+              InstanceKlass* buffered_klass =
+                ArchiveBuilder::current()->get_buffered_addr(InstanceKlass::cast(src_klass));
+              if (buffered_klass->has_aot_initialized_mirror()) {
+                st->print(" (aot-inited)");
+              }
+            }
+          }
+        }
```

Comment:
Is there a reason we wouldn't always print whether or not it is aot-inited? It seems useful information.